app/pages/challenges/components/challengeForm.tsx

```diff
@@ -46,23 +126,22 @@ export const ChallengeForm: React.FC<ChallengeFormProps> = ({
     setFormData((prev) => ({ ...prev, [name]: value }));
   };
 
-  const handleAddFriend = () => {
-    if (friendIds.trim()) {
-      setInvitedFriends((prev) => [...prev, friendIds.trim()]);
-      setFriendIds("");
-    }
+  const handleAddFriend = (friend: FriendOption) => {
+    setInvitedFriends((prev) => [...prev, friend]);
+    setIsSearchOpen(false);
+    setSearchTerm("");
   };
 
-  const handleRemoveFriend = (id: string) => {
-    setInvitedFriends((prev) => prev.filter((friend) => friend !== id));
+  const handleRemoveFriend = (id: number) => {
+    setInvitedFriends((prev) => prev.filter((friend) => friend.id !== id));
   };
```

Comment:
same here , invited friends should be fetched from server. use react query to fetch invited friends, when you are inviting a new one revalidate invited friends query to get new state